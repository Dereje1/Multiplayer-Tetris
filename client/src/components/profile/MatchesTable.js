import * as React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
/* MUI */
import Box from '@mui/material/Box';
import Collapse from '@mui/material/Collapse';
import IconButton from '@mui/material/IconButton';
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Typography from '@mui/material/Typography';
import Paper from '@mui/material/Paper';
import KeyboardArrowDownIcon from '@mui/icons-material/KeyboardArrowDown';
import KeyboardArrowUpIcon from '@mui/icons-material/KeyboardArrowUp';
import EmojiEventsIcon from '@mui/icons-material/EmojiEvents';
import SentimentVeryDissatisfiedIcon from '@mui/icons-material/SentimentVeryDissatisfied';

const getStatusIcon = (isWinner) => (
  isWinner ? <EmojiEventsIcon sx={{ color: 'green', fontSize: 26 }} />
    :
    <SentimentVeryDissatisfiedIcon sx={{ color: 'red', fontSize: 26 }} />
)

const extractRowInfo = ({ row, opponents, userId }) => {
  const isWinner = userId === row.winnerGoogleId
  const opponentId = isWinner ? row.looserGoogleId : row.winnerGoogleId
  const details = ['You', opponents[opponentId]].map((player, idx) => {
    if (idx === 0) {
      return isWinner ?
        {
          player,
          linesCleared: row.winnerLinesCleared,
          floorsRaised: row.winnerFloorsRaised,
          isDisqualified: 'No'
        }
        :
        {
          player,
          linesCleared: row.looserLinesCleared,
          floorsRaised: row.looserFloorsRaised,
          isDisqualified: row.looserDisqualified ? 'Yes' : 'No'
        }
    }

    return isWinner ?
      {
        player,
        linesCleared: row.looserLinesCleared,
        floorsRaised: row.looserFloorsRaised,
        isDisqualified: row.looserDisqualified ? 'Yes' : 'No'
      }
      :
      {
        player,
        linesCleared: row.winnerLinesCleared,
        floorsRaised: row.winnerFloorsRaised,
        isDisqualified: 'No'
      }
  })
  return {
    ...row,
    opponentName: opponents[opponentId],
    isWinner: getStatusIcon(isWinner),
    details
  }
}

export const Row = ({ row, opponents, userId }) => {
  const [open, setOpen] = React.useState(false);
  const rowInfo = extractRowInfo({ row, opponents, userId })
  return (
    <React.Fragment>
      <TableRow sx={{ '& > *': { borderBottom: 'unset' } }}>
        <TableCell>
          <IconButton
            aria-label="expand row"
            size="small"
            onClick={() => setOpen(!open)}
          >
            {open ? <KeyboardArrowUpIcon /> : <KeyboardArrowDownIcon />}
          </IconButton>
        </TableCell>
        <TableCell component="th" scope="row">
          {moment.utc(rowInfo.createdAt).format('MMMM Do YYYY, h:mm a')}
        </TableCell>
        <TableCell align="center">{rowInfo.opponentName}</TableCell>
        <TableCell align="center">{rowInfo.isWinner}</TableCell>
        <TableCell align="center">{rowInfo.difficulty}</TableCell>
      </TableRow>
      <TableRow>
        <TableCell style={{ paddingBottom: 0, paddingTop: 0 }} colSpan={6}>
          <Collapse in={open} timeout="auto" unmountOnExit>
            <Box sx={{ margin: 1 }}>
              <Typography  gutterBottom component="div" sx={{fontWeight:'bold'}}>
                Details
              </Typography>
              <Table size="small" aria-label="purchases">
                <TableHead>
                  <TableRow>
                    <TableCell></TableCell>
                    <TableCell align="center">Lines cleared</TableCell>
                    <TableCell align="center">Floors raised</TableCell>
                    <TableCell align="center">Disqualified</TableCell>
                  </TableRow>
                </TableHead>
                <TableBody>
                  {rowInfo.details.map((detailsRow, idx) => (
                    <TableRow key={`${rowInfo._id}_${idx}`}>
                      <TableCell component="th" scope="row">
                        {detailsRow.player}
                      </TableCell>
                      <TableCell align="center">{detailsRow.linesCleared}</TableCell>
                      <TableCell align="center">{detailsRow.floorsRaised}</TableCell>
                      <TableCell align="center">
                        {detailsRow.isDisqualified}
                      </TableCell>
                    </TableRow>
                  ))}
                </TableBody>
              </Table>
            </Box>
          </Collapse>
        </TableCell>
      </TableRow>
    </React.Fragment>
  );
}

export default function MatchesTable({ rows, opponents, userId }) {
  return (
    <TableContainer component={Paper}>
      <Table aria-label="collapsible table">
        <TableHead>
          <TableRow>
            <TableCell />
            <TableCell>Date (UTC)</TableCell>
            <TableCell align="center">Opponent</TableCell>
            <TableCell align="center">Win/Loss</TableCell>
            <TableCell align="center">Difficulty level</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row) => (
            <Row
              key={row._id}
              row={row}
              opponents={opponents}
              userId={userId}
            />
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

MatchesTable.propTypes = {
  rows:PropTypes.arrayOf(PropTypes.object).isRequired,
  opponents: PropTypes.object.isRequired,
  userId:PropTypes.string.isRequired
}

Row.propTypes = {
  row: PropTypes.object.isRequired,
  opponents: PropTypes.object.isRequired,
  userId:PropTypes.string.isRequired
}