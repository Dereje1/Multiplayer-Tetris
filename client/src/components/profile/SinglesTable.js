import * as React from 'react';
import PropTypes from 'prop-types';
import moment from 'moment';
/* MUI */
import Table from '@mui/material/Table';
import TableBody from '@mui/material/TableBody';
import TableCell from '@mui/material/TableCell';
import TableContainer from '@mui/material/TableContainer';
import TableHead from '@mui/material/TableHead';
import TableRow from '@mui/material/TableRow';
import Paper from '@mui/material/Paper';
import DeleteForeverIcon from '@mui/icons-material/DeleteForever';
import IconButton from '@mui/material/IconButton'

export default function SinglesTable({ rows, onDelete }) {
  return (
    <TableContainer component={Paper}>
      <Table sx={{ minWidth: 650 }} aria-label="simple table">
        <TableHead>
          <TableRow>
            <TableCell>Date</TableCell>
            <TableCell align="center">Lines Cleared</TableCell>
            <TableCell align="center">Level Reached</TableCell>
            <TableCell align="center" >Delete</TableCell>
          </TableRow>
        </TableHead>
        <TableBody>
          {rows.map((row) => (
            <TableRow
              key={row._id}
              sx={{ '&:last-child td, &:last-child th': { border: 0 } }}
            >
              <TableCell component="th" scope="row">
                {moment.utc(row.createdAt).format('MMMM Do YYYY, h:mm:ss a')}
              </TableCell>
              <TableCell align="center">{row.linesCleared}</TableCell>
              <TableCell align="center">{row.levelReached}</TableCell>
              <TableCell padding="checkbox" align='center'>
                <IconButton id="delete-result" onClick={() => onDelete(row._id)}>
                  <DeleteForeverIcon color='error' sx={{ fontSize: '1.3em' }} />
                </IconButton>
              </TableCell>
            </TableRow>
          ))}
        </TableBody>
      </Table>
    </TableContainer>
  );
}

SinglesTable.propTypes = {
  rows: PropTypes.arrayOf(PropTypes.object).isRequired,
  onDelete: PropTypes.func.isRequired
}