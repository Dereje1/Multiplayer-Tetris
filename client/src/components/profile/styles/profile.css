#profile {
  height: 80vh;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background: #5d95dd67;
}

#profile #card {
  width: 700px;
  height: 60%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background: black;
  -webkit-box-shadow: 7px -7px 0px 1px #8b9097;
          box-shadow: 7px -7px 0px 1px #8b9097;
  border-radius: 10px;
  font-family: 'Fugaz One', cursive;
  -webkit-animation: fadeprofile 1s linear 1;
          animation: fadeprofile 1s linear 1;
}

#profile #card #name {
  font-size: 2em;
  margin-top: 20px;
  font-weight: bold;
  color: #7ffff5eb;
}

#profile #card #detail {
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  font-size: 1em;
  width: 100%;
  height: 100%;
  border-top: 2px dashed #a29e9e;
  color: #df3a3a;
}

#profile #card #detail #match {
  width: 50%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  border-right: 2px dashed #a29e9e;
}

#profile #card #detail #match .head {
  color: #d075d0;
  font-size: 1.2em;
}

#profile #card #detail #match .recentmatches {
  border-top: 1px dotted #a29e9e;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  height: 100%;
  width: 100%;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  font-size: 0.8em;
}

#profile #card #detail #match .recentmatches .recentDescriptionHeader {
  -ms-flex-item-align: center;
      -ms-grid-row-align: center;
      align-self: center;
  color: #d075d0;
}

#profile #card #detail #match .recentmatches .recentDescription {
  margin-left: 20px;
  color: #bebfc8;
}

#profile #card #detail #match .recentmatches .info {
  cursor: pointer;
}

#profile #card #detail #match .recentmatches .info.Won {
  color: #2af92a;
}

#profile #card #detail #match .recentmatches .info.Lost {
  color: red;
}

#profile #card #detail #Single {
  width: 50%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
}

#profile #card #detail #Single .head {
  color: #eba03e;
  font-size: 1.2em;
}

#profile #card #detail #Single .recentgames {
  border-top: 1px dotted #a29e9e;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  height: 100%;
  width: 100%;
  -ms-flex-pack: distribute;
      justify-content: space-around;
  font-size: 0.8em;
}

#profile #card #detail #Single .recentgames .recentDescriptionHeader {
  -ms-flex-item-align: center;
      -ms-grid-row-align: center;
      align-self: center;
  color: #eba03e;
}

#profile #card #detail #Single .recentgames .recentDescription {
  margin-right: 20px;
  color: #bebfc8;
  -ms-flex-item-align: end;
      align-self: flex-end;
}

#profile #card #detail #Single .recentgames .info {
  cursor: pointer;
  color: #4ef0ff;
}

#profile #gameDetail {
  width: 400px;
  height: 60%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  -webkit-box-align: center;
      -ms-flex-align: center;
          align-items: center;
  background: black;
  border-radius: 10px;
  font-family: 'Fugaz One', cursive;
  color: #29f729;
  -webkit-animation: fadedetail 1s linear 1;
          animation: fadedetail 1s linear 1;
}

#profile #gameDetail .action {
  font-size: 30px;
  -ms-flex-item-align: end;
      align-self: flex-end;
  margin: 10px;
  cursor: pointer;
}

#profile #gameDetail .action.tweet {
  -ms-flex-item-align: end;
      align-self: flex-end;
  color: #fafeff;
  margin-top: auto;
}

#profile #gameDetail .detailitems {
  -ms-flex-item-align: start;
      align-self: flex-start;
  margin-left: 10px;
  margin-top: 10px;
}

@-webkit-keyframes fadedetail {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes fadedetail {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@-webkit-keyframes fadeprofile {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}

@keyframes fadeprofile {
  0% {
    opacity: 0;
  }
  100% {
    opacity: 1;
  }
}
/*# sourceMappingURL=profile.css.map */