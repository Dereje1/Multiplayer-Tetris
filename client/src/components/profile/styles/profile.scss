#profile{
    height: 80vh;
    display: flex;
    justify-content: center;
    align-items: center;
    background:#5d95dd67;
    #card{
        width: 700px;
        height: 60%;
        display: flex;
        flex-direction: column;
        align-items: center;
        background: black;
        box-shadow: 7px -7px 0px 1px #8b9097;
        border-radius: 10px;
        font-family: 'Fugaz One', cursive;
        animation: fadeprofile 1s linear 1;
        #name{
            font-size: 2em;
            margin-top: 20px;
            font-weight: bold;
            color: #7ffff5eb;
        }
        #detail{
            display: flex;
            font-size: 1em;
            width: 100%;
            height: 100%;
            border-top: 2px dashed #a29e9e;
            color: #df3a3a;
            #match{
                width:50%;
                display: flex;
                flex-direction: column;
                align-items: center;
                border-right: 2px dashed #a29e9e;
                .head{
                    color: #d075d0;
                    font-size: 1.2em;
                }
                .recentmatches{
                    border-top: 1px dotted #a29e9e;
                    display: flex;
                    flex-direction: column;
                    height: 100%;
                    width: 100%;
                    justify-content: space-around;
                    font-size:0.8em;
                    .recentDescriptionHeader{
                        align-self: center;
                        color: #d075d0;
                    }
                    .recentDescription{
                        margin-left: 20px;
                        color: #bebfc8;
                    }
                    .info{
                        cursor: pointer;
                        &.Won{
                            color: #2af92a
                        }
                        &.Lost{
                            color: red;
                        }
                    }
                }
            }
            #Single{
                width:50%;
                display: flex;
                flex-direction: column;
                align-items: center;
                .head{
                    color: #eba03e;
                    font-size: 1.2em;
                }
                .recentgames{
                    border-top: 1px dotted #a29e9e;
                    display: flex;
                    flex-direction: column;
                    height: 100%;
                    width: 100%;
                    justify-content: space-around;
                    font-size:0.8em;
                    .recentDescriptionHeader{
                        align-self: center;
                        color: #eba03e;
                    }
                    .recentDescription{
                        margin-right: 20px;
                        color: #bebfc8;
                        align-self: flex-end;
                    }
                    .info{
                        cursor: pointer;
                        color: #4ef0ff;
                    }
                }
            }
        }

    }
    #gameDetail{
        width: 400px;
        height: 60%;
        display: flex;
        flex-direction: column;
        align-items: center;
        background:black;
        border-radius: 10px;
        font-family: 'Fugaz One', cursive;
        color: #29f729;
        animation: fadedetail 1s linear 1;
        .action{
            font-size: 30px;
            align-self: flex-end;
            margin: 10px;
            cursor: pointer;
            &.tweet{
                align-self: flex-end;
                color: #fafeff;
                margin-top: auto;
            }
        }
        .detailitems{
            align-self: flex-start;
            margin-left: 10px;
            margin-top: 10px;
        }
    }
}

@keyframes fadedetail {
    0%{
      opacity: 0;
    }
    100%{
        opacity: 1;
    }
  }

@keyframes fadeprofile {
    0%{
        opacity: 0;
    }
    100%{
        opacity: 1;
    }
}