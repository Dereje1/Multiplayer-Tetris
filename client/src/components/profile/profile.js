import React from 'react';
import PropTypes from 'prop-types';
import RESTcall from '../../crud';
import { connect } from 'react-redux';
import SinglesTable from './SinglesTable';
import MatchesTable from './MatchesTable'
/* MUI */
import FormControl from '@mui/material/FormControl';
import FormGroup from '@mui/material/FormGroup';
import FormControlLabel from '@mui/material/FormControlLabel';
import Switch from '@mui/material/Switch';
import Box from '@mui/material/Box';
import Typography from '@mui/material/Typography';
import CircularProgress from '@mui/material/CircularProgress';

const mapStateToProps = ({ user }) => ({ user });

export class Profile extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      userData: null, // fetched from db
      mode: false // switch table mode
    };
  }

  componentDidMount() {
    this.fetchUserStats();
  }

  fetchUserStats = async () => {
    try {
      const userData = await RESTcall({ address: '/api/user' });
      this.setState({ userData });
    } catch {
      this.setState({ userData: null });
    }
  };

  deleteResult = (id) => {
    const { userData: { singleStats } } = this.state;
    const updatedSingleStats = singleStats.filter(stat => stat._id !== id)
    this.setState({
      userData: {
        ...this.state.userData,
        singleStats: updatedSingleStats
      }
    }, async () => await RESTcall({ address: `/api/delete_single/${id}`, method: 'delete' }))
  }

  render() {
    const { userData, mode } = this.state;
    const { user } = this.props;
    if (!userData) return (
      <Box sx={{
        marginLeft: 'auto',
        marginRight: 'auto',
        marginTop: 10,
        width: 900,
        height: 550,
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center'
      }}>
        <CircularProgress />
      </Box>
    );
    return (
      <Box
        sx={{
          marginLeft: 'auto',
          marginRight: 'auto',
          marginTop: 10,
          width: 900,
          height: 550,
          display: 'flex',
          flexDirection: 'column',
          alignItems: 'center'
        }}
      >
        <Typography sx={{ fontSize: 22 }} color="primary">
          {
            `Hi ${user.profile.displayname}`
          }
        </Typography>
        <FormControl component="fieldset" variant="standard" >
          <FormGroup>
            <FormControlLabel
              control={
                <Switch checked={mode} onChange={() => { this.setState({ mode: !mode }) }} name="display_mode" />
              }
              label={mode ? 'Matches' : 'Single games'}
            />
          </FormGroup>
        </FormControl>
        {
          mode ?
            <MatchesTable
              rows={userData.matchStats}
              opponents={userData.opponentNames}
              userId={user.profile.username}
            />
            :
            <SinglesTable rows={userData.singleStats} onDelete={this.deleteResult} />
        }
      </Box>
    );
  }

}

export default connect(mapStateToProps)(Profile);

Profile.defaultProps = {
  user: null,
};

Profile.propTypes = {
  user: PropTypes.shape({
    profile: PropTypes.shape({
      displayname: PropTypes.string,
      username: PropTypes.string,
    }),
  }),
};
