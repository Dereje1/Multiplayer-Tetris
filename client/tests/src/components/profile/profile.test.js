import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { Profile } from '../../../../src/components/profile/profile';
import RESTcall from '../../../../src/crud'
import { stubProfile } from '../../../stub'

jest.mock('../../../../src/crud')

describe('The profile component', () => {
    let props;
    beforeEach(() => {
        props = {
            user: { ...stubProfile.user }
        }
    })
    afterEach(() => {
        props = null
        RESTcall.mockClear()
    })
    test('will render for logged in user in single mode', async () => {
        RESTcall.mockImplementation(() => Promise.resolve(stubProfile.userProfileResponse ))
        const wrapper = shallow(<Profile {...props} />)
        await Promise.resolve()
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('will render for logged in user in match mode', async () => {
        RESTcall.mockImplementation(() => Promise.resolve(stubProfile.userProfileResponse ))
        const wrapper = shallow(<Profile {...props} />)
        await Promise.resolve()
        const tableSwitch = wrapper.find('ForwardRef(FormControlLabel)').props().control;
        tableSwitch.props.onChange()
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('will render progress bar if no data fetched yet or rejected', async () => {
        RESTcall.mockImplementation(() => Promise.reject(null))
        const wrapper = shallow(<Profile {...props} />)
        await Promise.resolve()
        expect(toJson(wrapper)).toMatchSnapshot();
    })
    test('will delete a single game result', async () => {
        RESTcall.mockImplementation(() => Promise.resolve(stubProfile.userProfileResponse ))
        const wrapper = shallow(<Profile {...props} />)
        await Promise.resolve()
        let singleIds = wrapper.state().userData.singleStats.map(d => d._id)
        expect(singleIds.includes('62e3f935616501f9bd180a71')).toBe(true)
        
        const deleteResult = wrapper.find('SinglesTable');
        deleteResult.props().onDelete('62e3f935616501f9bd180a71')
        singleIds = wrapper.state().userData.singleStats.map(d => d._id)
        expect(singleIds.includes('62e3f935616501f9bd180a71')).toBe(false)
        expect(RESTcall).toHaveBeenLastCalledWith({address: "/api/delete_single/62e3f935616501f9bd180a71", method: "delete"})
    })
})