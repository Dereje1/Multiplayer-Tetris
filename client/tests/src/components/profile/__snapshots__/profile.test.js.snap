// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The profile component will render for logged in user in match mode 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 550,
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 10,
      "width": 900,
    }
  }
>
  <ForwardRef(Typography)
    color="primary"
    sx={
      Object {
        "fontSize": 22,
      }
    }
  >
    Hi Stub Display Name
  </ForwardRef(Typography)>
  <ForwardRef(FormControl)
    component="fieldset"
    variant="standard"
  >
    <ForwardRef(FormGroup)>
      <ForwardRef(FormControlLabel)
        control={
          <ForwardRef(Switch)
            checked={true}
            name="display_mode"
            onChange={[Function]}
          />
        }
        label="Matches"
      />
    </ForwardRef(FormGroup)>
  </ForwardRef(FormControl)>
  <MatchesTable
    opponents={
      Object {
        "0002": "Stub oponent name",
      }
    }
    rows={
      Array [
        Object {
          "__v": 0,
          "_id": "62dd04cb1a1a257ead39a1ea",
          "createdAt": "2022-07-24T08:37:31.809Z",
          "difficulty": 2,
          "looserDisqualified": true,
          "looserFloorsRaised": 0,
          "looserGoogleId": "0002",
          "looserLinesCleared": 0,
          "updatedAt": "2022-07-24T08:37:31.809Z",
          "winnerFloorsRaised": 0,
          "winnerGoogleId": "0001",
          "winnerLinesCleared": 0,
        },
        Object {
          "__v": 0,
          "_id": "62dc633b2464e19567160cb3",
          "createdAt": "2022-07-23T21:08:11.812Z",
          "difficulty": 4,
          "looserDisqualified": false,
          "looserFloorsRaised": 0,
          "looserGoogleId": "0001",
          "looserLinesCleared": 0,
          "updatedAt": "2022-07-23T21:08:11.812Z",
          "winnerFloorsRaised": 1,
          "winnerGoogleId": "0002",
          "winnerLinesCleared": 2,
        },
      ]
    }
    userId="0001"
  />
</ForwardRef(Box)>
`;

exports[`The profile component will render for logged in user in single mode 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 550,
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 10,
      "width": 900,
    }
  }
>
  <ForwardRef(Typography)
    color="primary"
    sx={
      Object {
        "fontSize": 22,
      }
    }
  >
    Hi Stub Display Name
  </ForwardRef(Typography)>
  <ForwardRef(FormControl)
    component="fieldset"
    variant="standard"
  >
    <ForwardRef(FormGroup)>
      <ForwardRef(FormControlLabel)
        control={
          <ForwardRef(Switch)
            checked={false}
            name="display_mode"
            onChange={[Function]}
          />
        }
        label="Single games"
      />
    </ForwardRef(FormGroup)>
  </ForwardRef(FormControl)>
  <SinglesTable
    onDelete={[Function]}
    rows={
      Array [
        Object {
          "__v": 0,
          "_id": "62e3f935616501f9bd180a71",
          "createdAt": "2022-07-29T15:13:57.976Z",
          "googleId": "0001",
          "levelReached": 9,
          "linesCleared": 49,
          "updatedAt": "2022-07-29T15:13:57.976Z",
        },
        Object {
          "__v": 0,
          "_id": "62dfd050d218114e5c901c43",
          "createdAt": "2022-07-26T11:30:24.576Z",
          "googleId": "0001",
          "levelReached": 12,
          "linesCleared": 60,
          "updatedAt": "2022-07-26T11:30:24.576Z",
        },
      ]
    }
  />
</ForwardRef(Box)>
`;

exports[`The profile component will render progress bar if no data fetched yet or rejected 1`] = `
<ForwardRef(Box)
  sx={
    Object {
      "alignItems": "center",
      "display": "flex",
      "flexDirection": "column",
      "height": 550,
      "marginLeft": "auto",
      "marginRight": "auto",
      "marginTop": 10,
      "width": 900,
    }
  }
>
  <ForwardRef(CircularProgress) />
</ForwardRef(Box)>
`;
